Routers/Routers/Routers/Routers.cs

```diff
@@ -0,0 +1,290 @@
+                array.Add(second);
+            }
+            else if (second == nodeIndex)
+            {
+                array.Add(first);
+            }
+        }
+
+        foreach (int item in array)
+        {
+            if (!isTerminal.Contains(item))
+            {
+                NumberOfConnectedGraphElements(item, isTerminal);
+            }
+        }
+
+        return isTerminal;
+    }
+
+    /// <summary>
+    /// Function for checking the connectivity of a graph
+    /// </summary>
+    /// <returns>Is the graph connected</returns>
+    public bool IsTheGraphConnected()
+    => nodeValue.Count == 0 || NumberOfConnectedGraphElements(nodeValue[0], new()).Count == nodes.Count;
+
+    /// <summary>
+    /// Function for graph constructionПуть 
+    /// </summary>
+    /// <param name="pathToFile">The path to the file with the table</param>
+    public void BuildGraph(string pathToFile)
+    {
+        var stringToConvert = File.ReadAllLines(pathToFile);
+        for (int i = 0; i < stringToConvert.Length; i++)
+        {
+            var newStringArray = stringToConvert[i].Split(' ');
+            if (Int32.TryParse(newStringArray[0], out int firstNumber))
+            {
+                AddNode(firstNumber);
+            }
+
+            int secondNumber = 0;
+            for (int j = 1; j < newStringArray.Length; j++)
+            {
+                if (newStringArray[j] == ":")
+                {
+                    continue;
+                }
+
+                if (Int32.TryParse(newStringArray[j], out int number))
+                {
+                    AddNode(number);
+                    secondNumber = number;
+                }
+
+                else
+                {
+                    int k = 0;
+                    string newNumber = "";
+                    while (newStringArray[j][k] != ')')
+                    {
+                        if (newStringArray[j][k] == '(' || newStringArray[j][k] == ',')
+                        {
+                            k++;
+                            continue;
+                        }
+
+                        newNumber += newStringArray[j][k];
+                        k++;
+                    }
+
+                    if (Int32.TryParse(newNumber, out int length))
+                    {
+                        SetLength(firstNumber, secondNumber, length);
+                    }
+                }
+            }
+        }
+    }
+
+    /// <summary>
+    /// Function for graph printing
+    /// </summary>
+    /// <param name="pathToFile">File path</param>
+    public void PrintGraph(string pathToFile)
+    {
+        ConvertToAnotherDictionary();
+        using StreamWriter writer = new(pathToFile);
+        {
+            for (int i = 0; i < nodeValue.Count; i++)
+            {
+                if (nodes[i].newDictionary.Keys.Count != 0)
+                {
+                    writer.Write($"{nodeValue[i]} : ");
+                }
+
+                else
```

Comment:
```suggestion
                }
                else
```
Перед else пустая строка не ставится